gef.py

```diff
@@ -2561,7 +2574,7 @@ def is_branch_taken(self, insn) -> Tuple[bool, str]:
 
 @register_architecture
 class X86(Architecture):
-    aliases = ("X86", Elf.X86_32)
+    aliases: Tuple[Union[int, str], ...] = ("X86", Elf.X86_32)
```

Comment:
Why put a hint here, it should be covered by the parent class?

Why is the magic an alias here? Do we use it when parsing an elf to find the architecture?

Reply:
> Why put a hint here, it should be covered by the parent class?
yes, but X86 has a new subclass which doesn't preserve the original type hint but the concrete one of its super class (so of X86).
> Why is the magic an alias here? Do we use it when parsing an elf to find the architecture?
As I said in your last comment: I was wondering about that as well but didn't investigate further

Reply:
How is this type different? The original type is `Tuple[Union[str, int], ...]`, the only difference is the order of `str` vs `int`, but that doesn't matter in a union.

Reply:
The original type hint is `Tuple[Union[str, int], ...]` but the concrete type of Architecture `X86` is `Tuple[str, int]` which is inherited to the subclass instead of the original type hint...

Reply:
Oh, but why use that type hint? We should use the more generic type hint of the top class. We don't want the class hierarchy to have diverging types.

Reply:
I understand now. We can just repeat the type hint from `Architecture` here. Technically the types in the union are swapped, which doesn't matter, but we should make them match for grepping and stuff.

Reply:
So nit: saw str and int to match `Architecture` so that it's more clearly a copy-paste of the hint.

Reply:
ok, commited the nitpick. please resolve this comment